packages/fhir-router/src/repo.ts

```diff
@@ -409,6 +417,17 @@ export class MemoryRepository extends BaseRepository implements FhirRepository {
     };
   }
 
+  async searchByReference<T extends Resource>(
+    searchRequest: SearchRequest<T>,
+    referenceField: string,
+    references: string[]
+  ): Promise<Record<string, T[]>> {
+    const combinedFilter: Filter = { code: referenceField, operator: Operator.IN, value: references.join(',') };
+    searchRequest.filters ??= [];
+    searchRequest.filters.push(combinedFilter);
+    return {};
```

Comment:
Hm.. how hard would it be to implement this in `MemoryRepository`?

I suppose we could just write the brute force version, as this would really only be used in unit tests and Storybook graphql examples.

Reply:
I'm fairly certain this was actually working as-is in `MemoryRepository` since it already support matching on `Operator.IN`. I don't quite remember why I left this function not functional...probably because it was not on the critical path of testing out the SQL-based functionality which is all I was focused on at the time.